components/director/internal/authenticator/claims/validator.go

```diff
@@ -153,6 +177,77 @@ func (v *validator) validateRuntimeConsumer(ctx context.Context, claims Claims)
+		return errors.Wrapf(err, "An error has occurred while opening transaction")
+	}
+	defer v.transact.RollbackUnlessCommitted(ctx, tx)
+
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	if len(claims.TokenClientID) == 0 {
+		log.C(ctx).Errorf("Could not find consumer token client ID")
+		return apperrors.NewUnauthorizedError("could not find consumer token client ID")
+	}
+	if len(claims.Region) == 0 {
+		log.C(ctx).Errorf("Could not determine consumer token's region")
+		return apperrors.NewUnauthorizedError("could not determine token's region")
+	}
+
+	tokenClientID := strings.TrimPrefix(claims.TokenClientID, v.tokenPrefix)
+	filters := []*labelfilter.LabelFilter{
+		labelfilter.NewForKeyWithQuery(v.subscriptionProviderLabelKey, fmt.Sprintf("\"%s\"", tokenClientID)),
+		labelfilter.NewForKeyWithQuery(tenant.RegionLabelKey, fmt.Sprintf("\"%s\"", claims.Region)),
+	}
+
+	providerInternalTenantID := claims.Tenant[tenantmapping.ProviderTenantKey]
+	providerExternalTenantID := claims.Tenant[tenantmapping.ProviderExternalTenantKey]
+	ctxWithProviderTenant := tenant.SaveToContext(ctx, providerInternalTenantID, providerExternalTenantID)
+
+	log.C(ctx).Infof("Get application template in provider tenant %s for labels %s: %s and %s: %s", providerInternalTenantID, tenant.RegionLabelKey, claims.Region, v.subscriptionProviderLabelKey, tokenClientID)
+	applicationTemplate, err := v.appTemplateSvc.GetByFilters(ctxWithProviderTenant, filters)
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("Error while getting application template in provider tenant %s for labels %s: %s and %s: %s: %v", providerInternalTenantID, tenant.RegionLabelKey, claims.Region, v.subscriptionProviderLabelKey, tokenClientID, err)
+		return errors.Wrapf(err, "failed to get application template in tenant %s for labels %s: %s and %s: %s", providerInternalTenantID, tenant.RegionLabelKey, claims.Region, v.subscriptionProviderLabelKey, tokenClientID)
+	}
+	log.C(ctx).Infof("Found application template with ID %q in provider tenant %s for labels %s: %s and %s: %s", applicationTemplate.ID, providerInternalTenantID, tenant.RegionLabelKey, claims.Region, v.subscriptionProviderLabelKey, tokenClientID)
+
+	consumerInternalTenantID := claims.Tenant[tenantmapping.ConsumerTenantKey]
+	consumerExternalTenantID := claims.Tenant[tenantmapping.ExternalTenantKey]
+	ctxWithConsumerTenant := tenant.SaveToContext(ctx, consumerInternalTenantID, consumerExternalTenantID)
+
+	applicatinsFilter := []*labelfilter.LabelFilter{
+		labelfilter.NewForKeyWithQuery(v.consumerSubaccountLabelKey, fmt.Sprintf("\"%s\"", consumerExternalTenantID)),
+	}
+
+	found := false
+
+	log.C(ctx).Infof("Listing applications in the consumer tenant %q for application template with ID: %q and label with key: %q and value: %q", consumerExternalTenantID, applicationTemplate.ID, v.consumerSubaccountLabelKey, consumerExternalTenantID)
+	applicationsPage, err := v.applicationSvc.List(ctxWithConsumerTenant, applicatinsFilter, 100, "")
```

Comment:
Can you increase the page size to 200 here, so that we postpone running into the issue of missing out some application due to us not going over all pages

Reply:
btw, you can do the same for the `validateRuntimeConsumer` function